Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -349,24 +382,124 @@ public sealed class MemoryView : ICodeFormattable {
                 return _buffer.GetItem((_start / _itemsize) + (index * _step));
             }
 
-            int firstByteIndex = _start + index * (int)itemsize * _step;
-            object result = packBytes(format, getByteRange(firstByteIndex, (int)itemsize), 0, (int)_buffer.ItemSize);
+            int firstByteIndex = _start + index * _itemsize * _step;
+            object result = packBytes(format, getByteRange(firstByteIndex, _itemsize), 0, (int)_buffer.ItemSize);
 
-            // TODO: BigInteger should also be merged into an integer once the int/long merge occurs
-            if (result is BigInteger || result is double || result is float || result is Bytes) {
+            // TODO: BigInteger/long/ulong should also be merged into an integer once the int/long merge occurs
+            if (result is BigInteger || result is double || result is Bytes) {
                 return result;
+            } else if (result is float) {
+                return Converter.ConvertToDouble(result);
+            } else if (result is long || result is ulong) {
+                return Converter.ConvertToBigInteger(result);
             } else {
                 return Convert.ToInt32(result);
             }
         }
 
+        /// <summary>
+        /// Verifies that the value being set does not overflow the format
+        /// for this MemoryView.
+        /// </summary>
+        /// <param name="value">The value to be set.</param>
+        private void checkNumeric(BigInteger value) {
```

Comment:
`checkForOverflow`. Should this be in TypecodeOps?

Reply:
`q` and `Q` seem to be flipped here.

Reply:
Probably.

Might be lost in the noise, but it seems like we're doing a lot of converting back and forth: object -> BigInteger -> object -> primitive type -> byte[] -> primitive type -> object (and then SetItem does a bunch of the same work). Anyway, I guess we should get it working first before worrying about optimizations.

Reply:
Yeah I think removing some of the conversions here is a good idea, for the moment this conversion to BigInteger makes the overflow check easier.

Reply:
I was able to remove the conversion to BigInteger. Still thinking about the conversions to `long` and `ulong`, which are probably only necessary when we get a BigInteger value, to allow BitConverter to do what it wants.